lib/media/streaming_engine.js

```diff
@@ -329,14 +333,51 @@ shaka.media.StreamingEngine = class {
    *   reset the timestamp offset to ensure the new track segments are correctly
    *   placed on the timeline.
    */
-  switchVariant(
-      variant, clearBuffer = false, safeMargin = 0, force = false,
-      adaptation = false) {
+  async switchVariant(variant, clearBuffer = false,
+      safeMargin = 0, force = false, adaptation = false) {
+    this.previousVariant_ = this.currentVariant_;
     this.currentVariant_ = variant;
 
     if (!this.startupComplete_) {
       // The selected variant will be used in start().
       return;
+    } else {
+      // Audio codec change type
+      const ContentType = shaka.util.ManifestParserUtils.ContentType;
+      const Strategy = shaka.config.CodecSwitchingStrategy;
+      const changeRequired = (this.previousVariant_.audio.codecs !==
+        this.currentVariant_.audio.codecs);
+      const changeStrategy = this.config_.codecSwitchingStrategy;
+      const fullMimeType = shaka.util.MimeUtils.getFullType(
+          variant.audio.mimeType, variant.audio.codecs);
+
+      if (changeRequired) {
+        shaka.log.debug('Audio Codec Switch Required, using', changeStrategy);
+        switch (changeStrategy) {
+          case Strategy.DISABLED:
+            shaka.log.debug('Strategy DISABLED');
+            break;
+          case Strategy.SMOOTH:
+            shaka.log.debug('Strategy SMOOTH,' +
+             'attempting SourceBuffer.changeType');
+            await this.playerInterface_.mediaSourceEngine.changeType(
+                ContentType.AUDIO, fullMimeType);
+            break;
+          case Strategy.RELOAD:
+            shaka.log.debug('Strategy RELOAD');
+            this.unloadMediaStateByContentType(ContentType.AUDIO);
+            this.unloadMediaStateByContentType(ContentType.VIDEO);
+            await
```

Comment:
Did... the linter let you do this? The `this.playerInterface_.mediaSourceEngine` should be on the same line as the `await`.

Reply:
The linter does allow. I believe this was because of line length issues if same line but will double check.